### PR TITLE
Add new `gifts` field to Product type

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

Add a new `productGifts` field to Product type.

#### How should this be manually tested?

This field is being used by the ProductGifts component at:

[Workspace](https://productgifts--storecomponents.myvtex.com/gorgeous-watch/p)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: graphql/types/Product.graphql [hunk 1]
```diff
@@ -261,9 +261,33 @@ type Offer {
...
   teasers: [Teaser!]
   """
-  List of gifts associated with the product
+  List of SKUs for gifts associated with the product
   """
   giftSkuIds: [String]
+  """
+  List of gifts associated with the product
+  """
+  productGifts: [Gift]
```
Comment (klzns) [inline graphql/types/Product.graphql]: Why not just `gifts`?

File: graphql/types/Product.graphql [hunk 2]
```diff
@@ -261,9 +261,33 @@ type Offer {
...
+  linkText: String
+  description: String
+  productTitle: String
+  skuItem: GiftSkuItem
+}
+
+type GiftSkuItem {
+  nameComplete: String
+  images: [GiftImage]
+}
```
Comment (klzns) [inline graphql/types/Product.graphql]: Why not merge this into Gift type?

File: node/resolvers/search/offer.ts [hunk 3]
```diff
@@ -37,6 +37,45 @@ export const resolvers = {
     },
     teasers: propOr([], 'Teasers'),
     giftSkuIds: propOr([], 'GiftSkuIds'),
+    productGifts: ({ GiftSkuIds }: CommertialOffer, _: any, ctx: Context) => {
+      if (GiftSkuIds.length === 0) {
+        return []
+      }
+      const giftProducts = Promise.all(
```
Comment (klzns) [inline node/resolvers/search/offer.ts]: Add a new line here

File: graphql/types/Product.graphql (hunk 1 above)
Comment (victorhmp) [reply graphql/types/Product.graphql]: Could be hahaha

File: graphql/types/Product.graphql (hunk 2 above)
Comment (victorhmp) [reply graphql/types/Product.graphql]: No particular reason, guess those could be merged

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- New `productGifts` field to Product type.
```
Comment (klzns) [inline CHANGELOG.md]:
```suggestion
- New `gift` field to `Product` type.
```

File: graphql/types/Product.graphql [hunk 5]
```diff
@@ -261,9 +261,29 @@ type Offer {
...
+  gifts: [Gift]
+}
+
+type Gift {
+  nameComplete: String
+  productName: String
+  brand: String
+  linkText: String
+  description: String
+  productTitle: String
```
Comment (klzns) [inline graphql/types/Product.graphql]: What is this field? I think we should have only

```diff
- nameComplete
  productName
+ skuName
- productTitle
```

What do you think?

Comment (victorhmp) [reply graphql/types/Product.graphql]: I was thinking about using `productTitle` for the `gift-image` `` tag.
But I guess those two you pointed out could be merged and I do agree that `skuName` is a much better name.

**Pull request merged**